x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/bulk_actions/rule_params_modifier.ts

```diff
@@ -31,32 +30,32 @@ const applyBulkActionEditToRuleParams = (
   switch (action.type) {
     // index_patterns actions
     // index pattern is not present in machine learning rule type, so we throw error on it
-    case BulkActionEditType.add_index_patterns:
+    case BulkActionEditType.add_index_patterns: {
       invariant(
         ruleParams.type !== 'machine_learning',
         "Index patterns can't be added. Machine learning rule doesn't have index patterns property"
       );
 
-      if (ruleParams.dataViewId != null && !action.overwrite_data_views) {
-        break;
-      }
+      const dataViewExistsAndNotOverriden =
+        ruleParams.dataViewId != null && !action.overwrite_data_views;
+      invariant(!dataViewExistsAndNotOverriden, BulkEditSkipReason.DataViewExistsAndNotOverriden);
```

Comment:
I believe skipping shouldn't be implemented via throwing exceptions and then checking what exceptions we've got after calling `rulesClient.bulkEdit()`. Skipping should be a part of the normal business logic and should be represented in the code explicitly as one of the results.

Reply:
Reimplemented without exceptions.